pkgs/native_assets_cli/lib/src/model/link_output.dart

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of '../api/link_output.dart';
+
+class LinkOutputImpl implements LinkOutput {
+  final BuildOutputImpl _buildOutput;
```

Comment:
Composition doesn't seem like the right solution here. (It makes the buildoutput impl more first class than the link output.)

Can we make a `BuildStepOutputImpl` that just implements both? (And if not, make that a base class of LinkOutputImpl and BuildOutputImpl.)

Reply:
I would like to avoid duplicating code, especially around serialization/deserialization. I think composition solves this problem easily, as a `LinkOutput` has strictly less fields than a `BuildOutput`. If `LinkOutput` were to gain additional fields, then writing it's own serialization would make sense.

Reply:
I refactored it to have `LinkOutput` and `BuildOutput` to both be implemented by the same class.